osquery/tables/events/linux/apparmor_events.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+#include <osquery/registry_factory.h>
+
+#include "osquery/events/linux/auditeventpublisher.h"
+#include "osquery/tables/events/linux/apparmor_events.h"
+
+namespace osquery {
+FLAG(bool,
+     audit_allow_apparmor_events,
+     false,
+     "Allow the audit publisher to process audit events");
+
+REGISTER(AppArmorEventSubscriber, "event_subscriber", "apparmor_events");
+
+namespace tables {
+extern long getUptime();
+}
+
+Status AppArmorEventSubscriber::init() {
+  if (!FLAGS_audit_allow_apparmor_events) {
+    return Status(1, "Subscriber disabled via configuration");
+  }
+
+  auto sc = createSubscriptionContext();
+  subscribe(&AppArmorEventSubscriber::Callback, sc);
+
+  return Status(0, "Ok");
+}
+
+Status AppArmorEventSubscriber::Callback(const ECRef& ec, const SCRef& sc) {
+  std::vector<Row> emitted_row_list;
+  auto status = ProcessEvents(emitted_row_list, ec->audit_events);
+  if (!status.ok()) {
+    return status;
+  }
+
+  for (auto& row : emitted_row_list) {
+    add(row);
+  }
+
+  return Status(0, "Ok");
+}
+
+Status AppArmorEventSubscriber::ProcessEvents(
+    std::vector<Row>& emitted_row_list,
+    const std::vector<AuditEvent>& event_list) noexcept {
+  emitted_row_list.clear();
+
+  for (const auto& event : event_list) {
+    if (event.type != AuditEvent::Type::AppArmor) {
```

Comment:
Should this ever happen? Should there be a log here if it does?

Reply:
I added logging and fixed that place